src/java/htsjdk/samtools/Bin.java

```diff
@@ -76,8 +76,12 @@ public int getBinNumber() {
      */
     @Override
     public boolean equals(final Object other) {
-        if(other == null) return false;
-        if(!(other instanceof Bin)) return false;
+        if(other == null) {
```

Comment:
Can you please stick to single-line style `if`s for guard statements?  I.e.
```java
if (other == null) return false;`
```
is preferred over
```java
if (other == null) {
    return false;
}
```
especially when used to return early at the top of methods.